src/utils_db_query.c

```diff
@@ -1011,7 +1012,13 @@ int udb_query_prepare_result (udb_query_t const *q, /* {{{ */
   if ((q == NULL) || (prep_area == NULL))
     return (-EINVAL);
 
-  udb_query_finish_result (q, prep_area);
```

Comment:
Why did you change this? I didn't look at all details of the data flow here but `udb_query_finish_result` does more than deallocate fields of `prep_area`. It's generally a good idea to make sure you clean up after yourself, so keeping this call here even if it's a no-op (at the moment) is the better option imho. Otherwise, future, unrelated changes in other parts of this file may require changes here as well which should be avoided.